Fix ccflags order

mppf opened the pull request:
I realized that my PR #1592 changed the behavior of --ccflags, so that
the Chapel added flags were after the user ones. We want the Chapel
ones before the user flags. This patch fixes it.

mppf commented:
@thomasvandoren, I'm not sure I need review, but I wanted to give you a chance to object :)

thomasvandoren commented:
Can you add a regression test that verifies the original issue was fixed? I think clang's --debug vs --no-debug might work, assuming one cancels out the other depending on where it shows up.

mppf commented:
Sure, I'll add a test.

thomasvandoren commented:
This looks good, btw! Thanks!

mppf commented:
@thomasvandoren, this patch got a bit bigger. I added a test that specified multiple --cflags but then
discovered that didn't work, so I fixed that. While I was there, I realized that any string arguments we
handle now will not fail with an error if they are too long (at least we don't core dump though) so I added an error for that. Could you have another look?

mppf commented:
I'm still planning more testing... but the new tests work for me.

mppf commented:
Passed full testing

mppf commented:
@thomasvandoren, this patch is awaiting a quick look from you on the new changes since it does more stuff..

thomasvandoren commented on compiler/main/arg.cpp:
```diff
@@ -681,7 +681,14 @@ static void process_arg(const ArgumentState*       state,
+            int maxlen = atoi(desc->type + 1);
+            if( len > maxlen ) {
+              USR_FATAL("argument for %s is too long", desc->name);
+            }
```
:+1:

thomasvandoren replied:
What do you think about adding two dashes before the name? I.e.

```
USR_FATAL("argument for --%s is too long", desc->name);
```

mppf replied:
I'll do that.

thomasvandoren commented on compiler/main/driver.cpp:
```diff
@@ -454,6 +455,21 @@ static void setDevelSettings(const ArgumentState* state, const char* arg_unused)
   }
 }
 
+static void setCCFlags(const ArgumentState* state, const char* arg) {
```
Will you please add a comment describing what this does (and why it's necessary)? I understand after looking at it, but was originally confused especially since ccflagsArgument also exists.

thomasvandoren commented:
This looks good! I added a couple comments inline, but nothing major. Feel free to merge when you are ready.
